File: LeetCode-Python/MaximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = current_sum = nums[0]

        for num in nums[1:]:
            # Yeni alt dizi başlatmak mı, yoksa mevcut alt diziye devam etmek mi?
            current_sum = max(num, current_sum + num)
            # Şu ana kadarki en büyük toplamı güncelle
            max_sum = max(max_sum, current_sum)

        return max_sum

File: LeetCode-Cpp/MaximumSubarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = nums[0];
        int currentSum = nums[0];

        for (int i = 1; i < nums.size(); ++i) {
            // Yeni alt dizi başlatmak mı daha iyi, yoksa mevcut alt diziye devam etmek mi?
            currentSum = max(nums[i], currentSum + nums[i]);
            // Şu ana kadarki en büyük toplamı güncelle
            maxSum = max(maxSum, currentSum);
        }

        return maxSum;

    }
};
